### PR TITLE
Cherry Pick to 6.7: Preload: fix multiple regressions around global styles (#66468)

### PR DESCRIPTION
**Pull request by ramonjd**
Cherry picking https://github.com/WordPress/gutenberg/pull/66468 to `wp/6.7`

### Testing

Fire up this branch and head to the site editor.

Check that global-styles endpoints aren't fetched client side, but are, rather, preloaded via [rest_preload_api_request](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api.php#L2918)

Do the same for the post editor.

### CODE REVIEW

File: lib/compat/wordpress-6.8/preload.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Preload theme and global styles paths to avoid flash of variation styles in
+ * post editor.
+ *
+ * @param array                   $paths REST API paths to preload.
+ * @param WP_Block_Editor_Context $context Current block editor context.
+ * @return array Filtered preload paths.
+ */
+function gutenberg_block_editor_preload_paths_6_8( $paths, $context ) {
```
Comment (ramonjd) [inline lib/compat/wordpress-6.8/preload.php]: This will be moved to 6.7 compat.

Since it doesn't affect packages I think it can be done in a follow up?

Comment (ramonjd) [reply lib/compat/wordpress-6.8/preload.php]: https://github.com/WordPress/gutenberg/pull/66543

Comment (aaronrobertshaw) [reply lib/compat/wordpress-6.8/preload.php]: This was taken care of in this PR via https://github.com/WordPress/gutenberg/pull/66542/commits/287828afa2c1305d3e9bfa467d5defab759d50b0 and https://github.com/WordPress/gutenberg/pull/66542/commits/bc96921fff486fc4f44534698bb3f0db3808a775

**Review by andrewserong — APPROVED** [review]
This is testing well for me in the post and site editors, thanks for backporting!

LGTM 🚀

**Review by aaronrobertshaw — APPROVED** [review]
Thanks for wrangling this @ramonjd 👍 

✅ Code changes match combination of https://github.com/WordPress/gutenberg/pull/66468 and https://github.com/WordPress/gutenberg/pull/66543
✅ Global styles paths are preloaded in both site and post editors

| Current 6.7 | This PR |
|---|---|
| Screenshot 2024-10-29 at 12 25 30 pm | Screenshot 2024-10-29 at 12 23 31 pm |

LGTM 🚢

**Pull request merged**